### ISSUE TITLE
🐛 React lazy import path not being picked up 

### ISSUE THREAD

**Issue by ryan-tote** [webpro-nl/knip#839]
### Prerequisites

- [X] I'm using the latest version
- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've checked the [list of known issues](https://knip.dev/reference/known-issues)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://codesandbox.io/p/sandbox/ls97j5?file=%2Fsrc%2FApp.tsx%3A25%2C5

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Description of the issue

An ordinary react lazy where the import path is inside the function works fine.

```typescript
const Button = lazy(() => {
  return import("./components/Button").then((mod) => {
    return { default: mod.Button };
  });
});
```

But when you move the import path to a separate function it no longer follows the function path.

```typescript
const getComponentFile = () => {
  return import("./components/Button");
};

const Button = lazy(() => {
  return getComponentFile().then((mod) => {
    return { default: mod.Button };
  });
});

const preloadComponent = () => {
  void getComponentFile();
};
```

Knip believes that the Button component is not being used so it flags it up as an unused export

**Comment by webpro** [comment]
This is expected behavior.

Did you try your luck by adding `--include-libs`? Your CSB doesn't have a terminal so I can't try.

**Comment by ryan-tote** [comment]
Oh interesting, didn't know that.

I saw that dynamic imports don't work on the docs https://knip.dev/guides/handling-issues#dynamic-import-specifiers but this isn't exactly a dynamic import.

`--include-libs` does work but incurs a performance penalty. Does seem a bit weird that we need to add that flag since this isn't library code.

**Comment by webpro** [comment]
This mode is disabled by default, because it incurs a performance penalty.

Docs should be improved, indeed this isn't only about library code.

Glad it does seem to do the trick for you!

**Comment by webpro** [comment]
> `--include-libs` does work but incurs a performance penalty. Does seem a bit weird that we need to add that flag since this isn't library code.

Correction: what's added by `--include-libs` is library code so that the type definitions for e.g. `React.lazy` gets included.

What I forgot in my previous comment is that Knip is able to figure out that the named `Button` export of `"./components/Button"` is referenced (without the flag/type definitions).

```ts
  return import("./components/Button").then((mod) => {
    return { default: mod.Button };
  });
```

I've now fixed #855 which is related. With that, I'm going to close this issue.
